### ISSUE TITLE
change semantics of *set* options.

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#608]
There are a few options to sr3 that are of *set* type:  logEvents, and fileEvents.  Currently, when specifying the option, the additional events are added to the set (in boolean logic *or* operator applied.) for example:

* default:  set( [ 'create', 'delete', 'link', 'modify' ] )
* in the config file, *option create,link,modify* 
* result: no change... all three options present so or'd...

the likely intent is that the user was trying to remove delete from the set, and did not expect the union operator to be applied.  to get the desired behaviour one would set:

```
fileEvents None
fileEvents create,link,modify

```

Where the first option invocation resets to an empty set. and then the new items are unioned with the now empty set.   I think this is a violation of the https://en.wikipedia.org/wiki/Principle_of_least_astonishment

We should probably have the argument values entirely replace the existing value.

issue spawned from #605 

update 2023/01/26:

Note that there are two *set* type options extant, the other one being logEvents.  there are logEvents for every entry point in a flowcb callback, and some extras, so replacing the entire value is cumbersome for this case.  an ORing logic is more practical for this case... perhaps unit the two by adding an explicit sign of OR, and meet expectations for replacement by default.  

notation #1: plus-sign prefix
------------------------------------

I think prefixing the value set with a plus sign (+) is a relatively obvious way to denote union this with what is already there (the current behaviour.) so would add interpretation of the + sign to set option parsing, to retain both methods of setting values.
```

     logEvents +after_post

```
would give the current behaviour. where without the + it would replace all.

notation #2: current as a standin
-------------------------------------------

An alternative would be to have a value to standin as the current value...

```

   logEvents current,after_post

```
which only is an issue if "current" ever becomes a set value that you need.

so far, prefer 1st notation...

**Comment by petersilva** [comment]
I think it is a one line change in the option parsing logic in sr_config.   Most of the work would be in analsys for adapting the convert function in sr.py to correctly upgrade these cases.

-- if fileEvents None is not found prior to a fileEvents option, then need to change the setting to an or with the existing defaults... except... sr3 default is all events... so unless None is present, it has no effect, in which case, no change is required...

-- I don't think *delete* should be in the default for fileEvents... hmm... but it is already there in v2... so that would be introducing an incompatibility.

**Comment by petersilva** [comment]
@reidsunderland comments on + notation? is that too odd... suggestions?

**Comment by reidsunderland** [comment]
I like the + notation. The default behavior would be less confusing, but + allows us to keep the way it currently works.

**Comment by petersilva** [comment]
ok, so latest version, set semantics: 

* + add/union with current state.
* **None** empty the set
* **All** fill the set with all possible values
* - subtract/remove given set from the current set.

**Comment by petersilva** [comment]
included in 3.0.27
